lib/connect.js

```diff
@@ -170,7 +170,7 @@ async function connectThroughNode (c, node) {
     const addr = c.dht._sockets.remoteServerAddress()
 
     c.firewall = c.dht.firewalled ? FIREWALL.UNKNOWN : FIREWALL.OPEN
-    c.request = c.handshake.send({
+    c.request = await c.handshake.send({
```

Comment:
this needs an extra flag on c now, requesting: true|false, because we guard above using the c.request property if we should issue the handshake at all, and the await breaks that